### DISCUSSION TITLE
How to get CopilotChat to use github copilot and throw error if not authenticated?

### DISCUSSION THREAD

**Discussion by Coding-Journey** [CopilotC-Nvim/CopilotChat.nvim#258] [Q&A] [answered]
From reading the docs, Copilot.vim or zbirenbaum/copilot.lua is a 'dependency' plugin of CopilotChat, however I noticed that if you don't authenticate, CopilotChat still gets results, I then disabled the copilot.lua plugin and I'm still getting CopilotChat responses.

How can I ensure that:
- I specifically select the github copilot model
- I only receive responses if I have authenticated and github copilot is being used

Setup:
Windows 10, Alacritty, WSL, Tmux,
NVIM v0.10.0-dev-2787+g6cfca21ba
Build type: RelWithDebInfo
LuaJIT 2.1.1710088188

Branch: Canary

**Comment by jellydn** [comment]
Just FYI - There is a similar here https://github.com/CopilotC-Nvim/CopilotChat.nvim/issues/245

**Comment by deathbeam** [comment, marked as answer]
For authentication, all github copilot stuff uses same file for storing the token afaik, so if you are using for example copilot cli or even vscode, I think its stored in that file that we are reading as well. I am definitely sure that the CLI extension at least works like that. So if it is working for you then you must've authenticated through one of those methods and that stored your token.

You can see how the file is being loaded here:

https://github.com/CopilotC-Nvim/CopilotChat.nvim/blob/canary/lua/CopilotChat/copilot.lua#L85

And we will throw an error if it fails to load the token from there as well

**Reply by Coding-Journey** [reply]
Ah great, you're completely right, thanks for pointing out the code there :)

I deleted the hosts file and an error was thrown, along with being unable to use copilot chat, I re-authenticated and it worked again, that's given me confidence that it's using the official github copilot through my token

Thanks for your help!
